tasty/src/dotty/tools/tasty/TastyVersion.scala

```diff
@@ -0,0 +1,24 @@
+package dotty.tools.tasty
+
+case class TastyVersion(major: Int, minor: Int, experimental: Int) {
```

Comment:
If we're going to use this widely, I really think there should be a (global?) cache - no point having thousands of identical TastyVersion's and they could even be shared between parallel runs.

Reply:
or as was suggested previously, use an opaque type wrapping `Long`/`Int`

Reply:
Both those ideas sound good. I would add those optimizations in separate PR as this one is already a bit bloated.